datalad/support/sshconnector.py

```diff
@@ -118,6 +119,28 @@ def close(self):
             else:
                 raise
 
+    def copy(self, source, destination):
+        """Copies source file/folder to destination on the remote.
+
+        Parameters
+        ----------
+        source: str
+          file/folder path to copy from on local
+        destination: str
+          file/folder path to copy to on remote
+
+        Returns
+        -------
+        str
+          stdout, stderr of the copy operation.
+        """
+
+        self.scp_prefix = ["scp", "-o", "ControlPath=" + self.ctrl_path]
+        if isdir(source):  # add recursive flag if directory to be copied
+            self.scp_prefix = self.scp_prefix + ["-r"]
+        scp_cmd = self.scp_prefix + [source, self.host + ":" + destination]
```

Comment:
Please note, that the port needs to be included in `ctrl_path`. Originally it was handled already, when `urlparse` was used, since port was part of `netloc`. Apparently we missed to adapt when we switched to use `SSHRI`.